src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/Marshallers/NativeToManagedThisMarshallerFactory.cs

```diff
@@ -30,27 +31,34 @@ private sealed class Marshaller : IMarshallingGenerator
             public ManagedTypeInfo AsNativeType(TypePositionInfo info) => new PointerTypeInfo("void*", "void*", false);
             public IEnumerable<StatementSyntax> Generate(TypePositionInfo info, StubCodeContext context)
             {
+                TypeSyntax? unwrapperType = (context as NativeToManagedStubCodeContext)?.UnwrapperType;
+                Debug.Assert(unwrapperType != null);
```

Comment:
Instead of passing the unwrapper type through the context, can we pass it through the `NativeThisInfo` type (and make it not a singleton)? That way we wouldn't have a dependency on the specific underlying stub context type here.